src/pages/organize/[orgId]/projects/[campId]/areaassignments/[areaAssId]/report.tsx

```diff
@@ -94,216 +114,358 @@ const AreaAssignmentReportPage: PageWithLayout<AreaAssignmentReportProps> = ({
-                          gap={1}
-                          justifyContent="space-between"
-                          width="100%"
+
+                  {metricBeingDeleted?.definesDone && (
+                    <Box display="flex" flexDirection="column" gap={1}>
+                      <Typography>
+                        <Msg
+                          id={messagesIds.report.delete.deleteWarningText}
+                          values={{ title: metricBeingDeleted.question }}
+                        />
+                      </Typography>
+                      {assignment.metrics
+                        .filter(
+                          (metric) =>
+                            metric.kind == 'boolean' &&
+                            metric.id != metricBeingDeleted?.id
+                        )
+                        .map((metric) => (
+                          <Box
+                            key={metric.question}
+                            alignItems="center"
+                            display="flex"
+                            gap={1}
+                            justifyContent="space-between"
+                            width="100%"
+                          >
+                            {metric.question || messages.report.card.question()}
+                            <Button
+                              onClick={() => {
+                                if (metricBeingDeleted) {
+                                  const filtered = assignment.metrics.filter(
+                                    (metric) =>
+                                      metric.id != metricBeingDeleted.id
+                                  );
+                                  updateAreaAssignment({
+                                    metrics: [
+                                      ...filtered.slice(
+                                        0,
+                                        filtered.indexOf(metric)
+                                      ),
+                                      {
+                                        ...metric,
+                                        definesDone: true,
+                                      },
+                                      ...filtered.slice(
+                                        filtered.indexOf(metric) + 1
+                                      ),
+                                    ],
+                                  });
+                                }
+                                setAnchorEl(null);
+                                setMetricBeingDeleted(null);
+                              }}
+                              variant="outlined"
+                            >
+                              <Msg id={messagesIds.report.delete.select} />
+                            </Button>
+                          </Box>
+                        ))}
+                    </Box>
+                  )}
+
+                  {!metricBeingDeleted?.definesDone && (
```

Comment:
The box that renders here looks visually identical to the `ZUIConfirmDialog`, the component that we usually use for this type of flow. Is there a reason not to use the `ZUIConfirmDialog`? 

Also I think having this be one dialog that changes content depending on whether the selected metric "definesDone" or not, causes a slight visual bug when closing the dialog that makes the other version of the dialog blink by...

Reply:
Basically my suggestion would be to have one dialog that only contains the content for the flow of deleting the metric that defines done. So the `onClick` for the delete button would check if the metric "definesDone", if it does it would open this dialog, and if it does not it would use the `ZUIConfirmDialog` instead (for example of how to use it, search in the code for `ZUIConfirmDialogContext`)